### PR TITLE
Add modular AppHeader component

### PR DESCRIPTION
**Pull request by jamesricky**
This component sould be used to build a custom header for `MasterLayout`.

### CODE REVIEW

**Comment by nsams** [comment]
wow nice work, especially on the docs! :tada:

File: packages/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+function DefaultArrowDown(): React.ReactElement {
+    const { palette } = useTheme();
+    return <ChevronDown htmlColor={palette.primary.contrastText} />;
+}
+
+export function AppHeaderDropdown({
+    children,
+    actionContent,
```
Comment (rabengraph) [inline packages/admin/src/appHeader/dropdown/AppHeaderDropdown.tsx:30]: I am not sure if the term "content" is ideal. 
What about `actionChildren`, or `buttonChildren`? ...not really perfect either.

**Comment by rabengraph** [comment]
Is there a need for a general purpose Item `AppHeaderItem`, which can be put in the Header for anything custom, that cannot be built with AppHeaderDropdown, AppHeaderAction and the like?

Something similar to ToolBarItem in ToolBar https://github.com/vivid-planet/comet-admin/blob/next/packages/admin/src/common/toolbar/item/ToolbarItem.tsx

**Comment by nsams** [comment]
A few parts of this are similar or almost identical to the Toolbar. Have you considered sharing parts?

**Comment by jamesricky** [comment]
> A few parts of this are similar or almost identical to the Toolbar. Have you considered sharing parts?

Yes, I thought of making this by adding a `variant` prop to the toolbar to differentiate the styling.  
I came to the conclusion, that the styling and the use-case is too different, that would just add unnecessary complexity. 

Toolbar items like BackButton, Breadcrumbs and Title should not be needed in the Header, the only component that is very similar is the FillSpace. 

Maybe we could replace `AppHeaderFillSpace` and `ToolbarFillSpace` with a `FillSpace` in `admin/src/common/FillSpace`?
@manuelblum what do you think?

**Comment by nsams** [comment]
:+1:

just for FillSpace it's not worth complicating things

**Review by rabengraph — APPROVED** [review]

**Review by manuelblum — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Pull request merged**